### PR TITLE
fix: links

### LINKED ISSUE #257: bug: broken links
### Problem

Couple of broken links reported by the build command

### To reproduce

```
yarn && yarn build && npm run serve
```

- Access http://localhost:3000/
- Click on any of the 3 links at the bottom
image

- 404

Besides the other links that I did not check but I believe they might be broken as well.

### Expected behaviour

No broken links

### Screenshots/logs

Output from the build command:

```
[WARNING] Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken links found:

- On source page path = /cow-protocol:
   -> linking to cow-protocol/concepts (resolved as: /cow-protocol/concepts)
   -> linking to cow-protocol/tutorials (resolved as: /cow-protocol/tutorials)
   -> linking to cow-protocol/reference (resolved as: /cow-protocol/reference)

- On source page path = /cow-protocol/concepts/product-features/cow-hooks:
   -> linking to ../auctions/social-consensus (resolved as: /cow-protocol/concepts/auctions/social-consensus)

- On source page path = /cow-protocol/reference/core/auctions/the-problem:
   -> linking to social-consensus (resolved as: /cow-protocol/reference/core/auctions/social-consensus)

- On source page path = /cow-protocol/reference/core/intents/hooks:
   -> linking to ../auctions/social-consensus (resolved as: /cow-protocol/reference/core/auctions/social-consensus)

- On source page path = /cow-protocol/tutorials/arbitrate/orderbook:
   -> linking to ./driver (resolved as: /cow-protocol/tutorials/arbitrate/driver)

- On source page path = /cow-protocol/tutorials/arbitrate/solver/driver:
   -> linking to autopilot (resolved as: /cow-protocol/tutorials/arbitrate/solver/autopilot)

- On source page path = /cow-protocol/tutorials/arbitrate/solver/solver-engine:
   -> linking to ../../reference/apis/solver (resolved as: /cow-protocol/tutorials/reference/apis/solver)

- On source page path = /:
   -> linking to cow-protocol/concepts (resolved as: /cow-protocol/concepts)
   -> linking to cow-protocol/tutorials (resolved as: /cow-protocol/tutorials)
   -> linking to cow-protocol/reference (resolved as: /cow-protocol/reference)
```

### docs version/commit hash

deca94c262c06986b7ff46111b3057ddfb9ec63b

### PR DESCRIPTION
**Pull request by mfw78**
# Description

This PR fixes links between pages.

# Changes

- [x] Tighten up linking, now throw for `404`
- [x] Fix all links

## Related Issues

Fixes #257

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -25,6 +25,6 @@ CoW DAO is of its community, by its community, and for its community. CoW DAO us
 
 CoW DAO's documentation follows a "Concepts", "Procedures", "Technical Reference" methodology. For example when a user wants to learn:
 
-- **_What_** something is (example: batch auctions) → see [Concepts](cow-protocol/concepts)
-- **_How_** to do something (example: create an order) → see [Tutorials](cow-protocol/tutorials)
-- **_Technical_** information (example: such as SDKs, APIs) - see [Technical Reference](cow-protocol/reference)
+- **_What_** something is (example: batch auctions) → see [Concepts](/category/-concepts)
+- **_How_** to do something (example: create an order) → see [Tutorials](/category/-tutorials)
+- **_Technical_** information (example: such as SDKs, APIs) - see [Technical Reference](/category/-technical-reference)
```
Comment (alfetopito) [inline docs/README.md:30]: Why the dash prefix?

Comment (mfw78) [reply docs/README.md:30]: It's what worked? :see_no_evil:

Comment (alfetopito) [reply docs/README.md:30]: Haha sure, the question was more like: why is this needed? Is this any setting or a magic pattern on docusaurus?

Comment (mfw78) [reply docs/README.md:30]: I _think_ it's a way to get it to infer creating a layout when there's no root page that exists within a folder. IMO, each folder _should_ have some root page within it (which would remove this requirement).

**Pull request merged**